tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/provision/openshift/PostgreSQLImageOpenShiftProvisioner.java

```diff
@@ -60,13 +70,36 @@ protected void configureContainer(ContainerBuilder containerBuilder) {
 	@Override
 	public Map<String, String> getImageVariables() {
 		Map<String, String> vars = super.getImageVariables();
+		vars.remove(POSTGRESQL_USER);
+		vars.remove(POSTGRESQL_PASSWORD);
+		vars.remove(POSTGRESQL_ADMIN_PASSWORD);
+		vars.remove("POSTGRESQL_USERNAME");
 		vars.put("POSTGRESQL_MAX_CONNECTIONS", "100");
 		vars.put("POSTGRESQL_SHARED_BUFFERS", "16MB");
 		vars.put("POSTGRESQL_MAX_PREPARED_TRANSACTIONS", "90");
+		vars.put("POSTGRESQL_DATABASE", dbApplication.getDbName());
 		// Temporary workaround for https://github.com/sclorg/postgresql-container/issues/297
 		// Increase the "set_passwords.sh" timeout from the default 60s to 300s to give the
 		// PostgreSQL server chance properly to start under high OCP cluster load
 		vars.put("PGCTLTIMEOUT", "300");
 		return vars;
 	}
+
+	@Override
+	public void customizeApplicationBuilder(ApplicationBuilder appBuilder) {
+		// the secret is supposed to be used by applications connecting to the database
+		appBuilder.secret(getSecretName())
+				.setType(SecretType.OPAQUE)
+				.addData(POSTGRESQL_USER_KEY, dbApplication.getUser().getBytes())
+				.addData(POSTGRESQL_PASSWORD_KEY, dbApplication.getPassword().getBytes())
+				.addData(POSTGRESQL_ADMIN_PASSWORD_KEY,
+						dbApplication.getAdminPassword().getBytes());
+		// the secret is also used to configure POSTGRESQL_USER, POSTGRESQL_PASSWORD, POSTGRESQL_ADMIN_PASSWORD
+		appBuilder.deploymentConfig().podTemplate().container().configFromConfigMap(
+				getSecretName(),
+				(String t) -> t.replace("-", "_").toUpperCase(),
```

Comment:
It seems to me the logic in here is depending on external assumptions. What happens in case the logic  e.g.: here https://github.com/Intersmash/intersmash/pull/57/files#diff-fef49e5f13fc3b4dd73b6cc9fc1ee5935ea5e9b787e6a88e3a58bada547e714aR35 changes?

I guess we'll have failures until we fix this similarly. So what about isolating the logic for computing keys from values and values from keys? Or maybe can we simply use the values here?

Or maybe I am just missing the point, feel free to elaborate on this.

Reply:
I actually crafted this code with exact opposite purpose: set the values of 
```
	public static final String POSTGRESQL_USER = "POSTGRESQL_USER";
	public static final String POSTGRESQL_PASSWORD = "POSTGRESQL_PASSWORD";
	public static final String POSTGRESQL_ADMIN_PASSWORD = "POSTGRESQL_ADMIN_PASSWORD";
```
in one place and have all other places to be automatically adjusted, e.g. `KeycloakRealmImportOperatorProvisionerTest`;

That transformation `(String t) -> t.replace("-", "_").toUpperCase()` is actually the cause of this because OpenShift doesn't accept e.g. `POSTGRESQL_USER` as the key value in e.g.:

```
spec:
  db:
    database: keycloak
    host: postgresql
    passwordSecret:
      key: postgresql-password
      name: postgresql
    port: 5432
    usernameSecret:
      key: postgresql-user
      name: postgresql
    vendor: postgres
```

Reply:
Ok, I see but IMO it might have been useful to encapsulate the logic (upper2lower + replace and lower2upper + replace) in 2 static methods.

Reply:
Resolving. Not such a great priority right now, since all the logic belongs to the same class.